framework/src/main/java/org/checkerframework/framework/util/typeinference8/util/Resolution.java

```diff
@@ -213,36 +213,64 @@ private BoundSet resolveSmallestSet(Set<Variable> as, BoundSet boundSet) {
   }
 
   /**
-   * Resolves all the non-capture variables in {@code variables}.
+   * Resolves all the non-capture variables in {@code vars}.
    *
-   * @param variables the variables
+   * @param vars the variables
    */
-  private void resolveNonCapturesFirst(List<Variable> variables) {
-    Variable smallV;
+  private void resolveNonCapturesFirst(List<Variable> vars) {
+    // Variables that are not captures and need to be resolved. (Avoid side-effecting vars)
+    List<Variable> variables = new ArrayList<>(vars);
+    variables.removeIf(Variable::isCaptureVariable);
+    applyInstantiationsToBounds(variables);
+    variables.removeIf(v -> v.getBounds().hasInstantiation());
+
+    // Until variables is empty:
+    // Find the variable, alpha, in `variables` that has the fewest variables (that do not already
+    // have an instantiation) mentioned in alpha's bounds.
+    // Resolve alpha using the "noncapture" resolution method.
```

Comment:
What is the relationship between "instantiate" and "resolve"?

Reply:
Resolve means to find an instantiation for a variable using one of the resolution algorithms.